ni_measurementlink_service/_internal/discovery_client.py

```diff
@@ -204,12 +206,63 @@ def resolve_service(self, provided_interface: str, service_class: str = "") -> S
 
 def _get_discovery_service_address() -> str:
     key_file_path = _get_key_file_path()
+    _ensure_discovery_service_started(key_file_path)
     _logger.debug("Discovery service key file path: %s", key_file_path)
     with _open_key_file(str(key_file_path)) as key_file:
         key_json = json.load(key_file)
         return "localhost:" + key_json["InsecurePort"]
 
 
+def _ensure_discovery_service_started(key_file_path: pathlib.Path):
+    """Make sure discovery service is started.
+
+    Check whether discovery service key file exists, if not start the discovery
+    service.
+
+    """
+    if _key_file_exists(key_file_path):
+        return
+
+    exe_file_path = _get_discovery_service_location()
+    _start_service(exe_file_path)
+
+
+def _get_discovery_service_location() -> pathlib.PurePath:
+    """Gets the location of the discovery service process executable.
+
+    Returns:
+    -------
+        exe_file_path: path to the discovery service exe which can be used
+                to activate the discovery service.
+
+    """
+    registration_json_path = _get_registration_json_file_path()
+    registration_json_obj = json.load(open(str(registration_json_path)))
+    root_directory = pathlib.PurePath(registration_json_path).parent
+    exe_file_path = root_directory.joinpath(registration_json_obj["discovery"]["path"])
+    return exe_file_path
+
+
+def _get_registration_json_file_path():
+    return (
+        pathlib.Path(os.environ["ProgramFiles"])
```

Comment:
`%ProgramFiles%` evaluates to `C:\Program Files (x86)` in 32-bit processes. Use `%ProgramW6432%` instead.
```suggestion
        pathlib.Path(os.environ["ProgramW6432"])
```